src/main/kotlin/com/evalify/evalifybackend/semester/controller/SemesterController.kt

```
@@ -1,31 +1,106 @@
+        )
+    }
+
+    @PostMapping("/{id}/courses")
+    fun createCourseForSemester(
+        @PathVariable id: UUID,
+        @RequestBody courseRequest: CreateCourseRequest
+    ): ResponseEntity<CourseResponse> {
+        return try {
+            val course = semesterService.createCourseForSemester(id, courseRequest)
+            ResponseEntity(course, HttpStatus.CREATED)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)